.github/workflows/cherry-pick-since-release-branch.yml

```diff
@@ -78,8 +105,8 @@ jobs:
         uses: risingwavelabs/github-action-cherry-pick@master
```

Comment:
When manually triggered on main, we need to `actions/checkout@v4` the specific commit created by merging the PR, rather than the default (which is `main`).

Reply:
Addressed. Also tested in https://github.com/BugenZhao/risingwave

- Labeled before merged: https://github.com/BugenZhao/risingwave/pull/12
- Labeled after merged: https://github.com/BugenZhao/risingwave/pull/18
- Manually triggered: https://github.com/BugenZhao/risingwave/pull/21

Reply:
Maybe I mislead you to `actions/checkout@v4`. It seems `risingwavelabs/github-action-cherry-pick@master` is harded coded to `GITHUB_SHA`. Can you test cherry-picking an older PR not at the HEAD of main?

Reply:
You're pretty right... Changes:

- https://github.com/risingwavelabs/github-action-cherry-pick/pull/11
- [`7582246` (#21274)](https://github.com/risingwavelabs/risingwave/pull/21274/commits/7582246923a43b18096d9c6ac3761fa73c1c821a)

Now it finally works:
- https://github.com/BugenZhao/risingwave/pull/27